lib/Sabberworm/CSS/CSSList/CSSList.php

```diff
@@ -110,15 +111,34 @@ private static function parseAtRule(ParserState $oParserState) {
 			$oLocation = URL::parse($oParserState);
 			$oParserState->consumeWhiteSpace();
 			$sMediaQuery = null;
-			if (!$oParserState->comes(';')) {
-				$sMediaQuery = $oParserState->consumeUntil(';');
+			try {
+				if (!$oParserState->comes(';')) {
+					$sMediaQuery = $oParserState->consumeUntil(';');
```

Comment:
Maybe we can add an optional parameter to `consumeUntil` that allows reading till either the desired char or EOF, whichever comes first. Would clean up the code somewhat and also resolve using exceptions for control flow.

Reply:
Or simply add a sentinel value (e.g. `ParserState::EOL = 'EOL'`) that can be used like a regular token:

```php
$sMediaQuery = $oParserState->consumeUntil(array(';', ParserState::EOL));
```

Reply:
I agree, however `consumeUntil` takes a lot of parameters already, so I added a new constant `ParserState::EOF` which can be passed as the first argument of `consumeUntil`. Also helps with readability.

Reply:
LoL, sorry for the repeat, I posted my reply without realizing you had a second post. I set the value of `ParserState::EOF` to null though. Do you think this can cause issues?

Reply:
It seems like we had the same idea…

I think any value that isn’t a valid return value from `consume(1)` is fine so `null` should work.